src/components/ChatEntry.js

```diff
@@ -1,22 +1,59 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js';
+import { useState } from 'react';
 
 const ChatEntry = (props) => {
+  // debugger;
+  const [emoji, setEmoji] = useState(props.emoji);
+
+  const updateEmojiRed = () => {
+    setEmoji('❤️');
+  };
+
+  const updateEmojiWhite = () => {
+    setEmoji('🤍');
+  };
+
+  const updateEmoji = () => {
+    props.updateLikesCallback(props.id);
+    console.log('inside emoji');
+    if (props.liked === true) {
+      updateEmojiWhite();
+    } else {
+      updateEmojiRed();
+    }
+    console.log(props.liked);
+  };
+
+  let bubbleSender = 'chat-entry local';
+  if (props.sender === props.senderRemote) {
+    bubbleSender = 'chat-entry remote';
+  }
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={bubbleSender}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp} />
+        </p>
+        <button className="like" onClick={updateEmoji}>
+          {emoji}
```

Comment:
If you remove the emoji state variable, you have a few choices here.  One option is to use a ternary operator: `props.liked ? '❤️' : '🤍'`.  Another option is to write a function that returns the correct emoji based on the value of props.liked and call the function here.